Plugin/GuestPaymentInformationResetOrderId.php

```diff
@@ -65,9 +65,7 @@ public function beforeSavePaymentInformationAndPlaceOrder(
             $quoteIdMask = $this->quoteIdMaskFactory->create()->load($cartId, 'masked_id');
             $quoteId = $quoteIdMask->getQuoteId();
             $quote = $this->quoteRepository->get($quoteId);
-            if (preg_match('/^adyen_(?!pos_cloud$)/', strval($quote->getPayment()->getMethod()))) {
-                $quote->setReservedOrderId(null);
-            }
+            $quote->setReservedOrderId(null);
```

Comment:
Removing previous ugly regex is nice but we need put something in place of this. Otherwise, this interceptor will be run for any other payment service provider's module. [isAdyenMethod()](https://github.com/Adyen/adyen-magento2/blob/c8ba5d03e1f588a07e1ff539b15c05e6c6b98571/Helper/PaymentMethods.php#L127) can be used for this purpose instead of regex.